### ISSUE TITLE
Spurious missplaced-attribute warning with stop-after typing in OCaml 5.2.1

### ISSUE THREAD

**Issue by gcluzel** [ocaml/ocaml#13630]
The compilation of the following program with OCaml 5.2.1 with the option  `-stop-after typing` reports a false positive `missplaced-attribute` warning.

```ocaml
let rec f x =
  if x = 0 then x
  else (f [@tailcall]) (x - 1)
```

```shell
$ ocamlc -c test.ml
$ ocamlc -c -stop-after typing test.ml
3 |   else (f [@tailcall]) (x - 1)
                ^^^^^^^^
Warning 53 [misplaced-attribute]: the "tailcall" attribute cannot appear in this context
```

**Comment by nojb** [comment]
This was fixed in #13203, but the fix was not included in 5.2.1.

**Comment by gcluzel** [comment]
I didn't find this issue. Thanks for the pointer and sorry for the noice!
